nepalingo-web/src/components/user_auth/User_auth.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from "react";
+import supabase from "./supabaseClient";
+import { useNavigate } from "react-router-dom";
+
+const User_auth: React.FC = () => {
+  const [action, setAction] = useState<"Sign Up" | "Log In">("Sign Up");
+  const [username, setUsername] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [error, setError] = useState<string | null>(null);
+  const [success, setSuccess] = useState<boolean>(false);
+  const [showPassword, setShowPassword] = useState(false);
+  const navigate = useNavigate();
+
+  const handleSubmit = async () => {
+    setError(null);
+    setSuccess(false);
+
+    if (action === "Sign Up") {
```

Comment:
Add some comments and explain what's happening

Reply:
done